controllers/etcd_controller.go

```diff
@@ -577,6 +635,18 @@ func (r *EtcdReconciler) syncStatefulSetSpec(ss *appsv1.StatefulSet, cm *corev1.
 	ssCopy := ss.DeepCopy()
 	ssCopy.Spec.Replicas = decoded.Spec.Replicas
 	ssCopy.Spec.UpdateStrategy = decoded.Spec.UpdateStrategy
+
+	// Applying suggestions from
+	containers := getContainerMapFromPodTemplateSpec(ssCopy.Spec.Template.Spec)
```

Comment:
I didn't get this? what are we trying to achieve here? i assumed this was backward compatibility code first which is trying to sync `resource` requirement from existing sts to the sts which will be deployed as reconcile part of `etcd` crd resource.  But ideally that should be `etcd` crd deployers job and not of etcd-druid.  On the same line, since here etcd-druid is trying to do it on its own, what will happen to resource configured in etcd crd. Those are not propagated to etcd then.
I might have still missed something from druid working related to backward compatibility in regards to gardener. Please correct me in that case.

Reply:
This logic here is till etcd resouce can be updated by HVPA. Till then we will need druid to not override the changes applied by HVPA with the values in resources.